packages/grafana-runtime/src/services/dataSourceSrv.ts

```diff
@@ -33,6 +35,16 @@ export interface DataSourceSrv {
    * Reloads the DataSourceSrv
    */
   reload(): void;
+
+  /**
+   * Registers a runtime data source that can be used from scene applications
+   * Make sure your data source uid is unique
+   */
+  registerRuntimeDataSource(entry: RuntimeDataSourceRegistration): void;
```

Comment:
Also here wonder if this couldn't just be called `registerDataSource` and also drop the `that can be used from scene applications` from the comment. Like I assume it can be used anywhere.

Reply:
I will update the comment, good catch. Think the runtime data source naming is useful to separate it from a standalone data source plugin

Reply:
Yeah in the context of this https://github.com/grafana/grafana/pull/93956#discussion_r1842491722 the naming makes more sense.